roxie/roxiemem/roxiemem.cpp

```diff
@@ -1099,6 +1107,138 @@ bool HeapletBase::hasDestructor(const void *ptr)
         return false;
 }
 
+//================================================================================
+
+#ifdef _USE_CPPUNIT
+static size_t parallelSyncReleaseGranularity = DEFAULT_PARALLEL_SYNC_RELEASE_GRANULARITY;
+static size_t parallelSyncReleaseThreshold = DEFAULT_PARALLEL_SYNC_RELEASE_THRESHOLD;
+
+//Only used by the tuning unit tests, should possibly use a different flag to fix them as constant
+static void setParallelSyncReleaseGranularity(size_t granularity, unsigned scaling)
+{
+    parallelSyncReleaseGranularity = granularity;
+    parallelSyncReleaseThreshold = granularity * scaling;
+}
+#else
+const static size_t parallelSyncReleaseGranularity = DEFAULT_PARALLEL_SYNC_RELEASE_GRANULARITY;
+const static size_t parallelSyncReleaseThreshold = DEFAULT_PARALLEL_SYNC_RELEASE_THRESHOLD;
+#endif
+
+
+inline void inlineReleaseRoxieRowArray(size_t count, const void * * rows)
+{
+    for (size_t i = 0; i < count; i++)
+        ReleaseRoxieRow(rows[i]);
+}
+
+#ifdef PARALLEL_SYNC_RELEASE
+class sync_releaser_task : public tbb::task
+{
+public:
+    sync_releaser_task(size_t _num, const void * * _rows) : num(_num), rows(_rows) {}
+
+    virtual task * execute()
+    {
+        inlineReleaseRoxieRowArray(num, rows);
+        return NULL;
+    }
+
+public:
+    size_t num;
+    const void * * rows;
+};
+
+void ParallelReleaseRoxieRowArray(size_t count, const void * * rows)
+{
+    //Two different implementations of the same code.  The task list is simpler and the recommended approach, but
+    //appears to be slower - possibly because work doesn't start until all tasks created.  Needs more testing to verify.
+#if 1
+    tbb::task * completed_task = new (tbb::task::allocate_root()) tbb::empty_task();
```

Comment:
(1) What would you use numReleaseRowTasks for?

(2) This code doesn't create them all up front.  The code in the #else that uses the task list does, and seems to go slower because of it.